src/clustering/administration/servers/auto_reconnect.cc

```diff
@@ -84,10 +84,25 @@ void auto_reconnector_t::try_reconnect(const server_id_t &server,
 
     wait_any_t interruptor(&reconnected, &give_up_timer, keepalive.get_drain_signal());
 
+    cond_t join_failed;
     exponential_backoff_t backoff(50, 15 * 1000);
     try {
-        while (!interruptor.is_pulsed()) {
-            connectivity_cluster_run->join(last_known_address, join_delay_secs);
+        while (!interruptor.is_pulsed() && !join_failed.is_pulsed()) {
+            coro_t::spawn_now_dangerously([&]() {
```

Comment:
iirc this was an issue with the initial implementation where if I didn't specify the drainer of the `connectivity_cluster_run` then I got all sorts of weird threading issues. I'll give this a shot, but this just occurred to me.